File: app/src/main/java/ru/dmitrybochkov/rates/di/UseCasesModule.java
package ru.dmitrybochkov.rates.di;

import javax.inject.Singleton;

import androidx.annotation.NonNull;
import dagger.Module;
import dagger.Provides;
import ru.dmitrybochkov.rates.data.repository.Repository;
import ru.dmitrybochkov.rates.usecases.rates.RatesUseCase;

/**
 * Created by Dmitry Bochkov on 14.11.2019.
 */

@Module
public class UseCasesModule {

    @Provides
    @Singleton
    RatesUseCase provideRatesUseCase(@NonNull final Repository repository) {
        return new RatesUseCase(repository);
    }
}

File: app/src/main/java/ru/dmitrybochkov/rates/di/UseCasesModule.kt
package ru.dmitrybochkov.rates.di

import dagger.Module
import dagger.Provides
import ru.dmitrybochkov.rates.data.repository.Repository
import ru.dmitrybochkov.rates.usecases.rates.RatesUseCase
import javax.inject.Singleton

/**
 * Created by Dmitry Bochkov on 14.11.2019.
 */

@Module
class UseCasesModule {

    @Provides
    @Singleton
    internal fun provideRatesUseCase(repository: Repository): RatesUseCase {
        return RatesUseCase(repository)
    }
}
